### PR TITLE
Add option to collect cpu usage from processes

### PR DESCRIPTION
**Pull request by RMacfarlane**
Calculates the cpu usage of a process as the change in kernel + user time the process has used over the change in kernel + user time the system has used after sleeping for a second.

Currently this is being calculated for all processes when the flag is set instead of just the processes under rootId, as I was having trouble serializing the structure of cpu information back and forth between javascript and C++.

### CODE REVIEW

File: typings/windows-process-tree.d.ts
Comment (Tyriar) [inline typings/windows-process-tree.d.ts]: I think the API should look something like this (after the TS change is merged):

```ts
export interface ProcessCpuTreeNode extends ProcessTreeNode {
  cpu: number
};
export function getProcessCpuUsage(tree: ProcessTreeNode, callback: (tree: ProcessCpuTreeNode): void;
```

That way we don't need to complicate `getProcessTree` and everything is separated nicely. Also the small amount of time needed to cross the native boundary doesn't matter since this is a long running async operation.

File: test.js
Comment (Tyriar) [inline test.js]: Maybe checking `typeof cpu` and 0 <= cpu <= 100 would be a better check?

File: src/process.cc
Comment (Tyriar) [inline src/process.cc]: I think this needs some comments explaining how the CPU time calculation works.

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/process.h
Comment (RMacfarlane) [inline src/process.h]: Remove CPU flag

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/addon.cc
Comment (Tyriar) [inline src/addon.cc]: Is this being used?

File: typings/windows-process-tree.d.ts
Comment (Tyriar) [inline typings/windows-process-tree.d.ts]: Let's start filling in the JS Doc here as the API is getting a little more complex

**Review by Tyriar — APPROVED** [review]
LGTM after the 2 remaining comments, great job 😃

**Pull request merged**
